packages/server/src/studentTaskProgress/studentTaskProgress.controller.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  AllStudentAssignmentProgress,
+  Role,
+  StudentAssignmentProgress,
+  UserPartial,
+} from '@koh/common';
+import {
+  Controller,
+  UseGuards,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  Get,
+  Param,
+} from '@nestjs/common';
+import { Roles } from 'decorators/roles.decorator';
+import { JwtAuthGuard } from 'guards/jwt-auth.guard';
+import { Connection, getRepository } from 'typeorm';
+import { StudentTaskProgressModel } from './studentTaskProgress.entity';
+import { CourseRolesGuard } from 'guards/course-roles.guard';
+import { QueueRolesGuard } from 'guards/queue-role.guard';
+
+@Controller('studentTaskProgress')
+@UseGuards(JwtAuthGuard, CourseRolesGuard)
+@UseInterceptors(ClassSerializerInterceptor)
+export class StudentTaskProgressController {
+  constructor(private connection: Connection) {}
+  // TODO: make it so students can only retrieve their own taskProgress
+  // maybe make 3 endpoints. getMyAssignmentProgress (using @UserId), getStudentAssignmentProgress (only usable by TAs. Though, maybe instead just ship the assignment progress with the task question itself), and getAllAssignmentProgressForQueue (also only used by TAs. This can be called initially to save a lot of the initial calls.), can also have getAllAssignmentProgressForCourse
+  // will probably want to move the method to its own controller at that point (with a little note that mentions that the setting of taskProgress happens in the updateQuestion endpoint in the question controller).
+  @Get('student/:userId/:courseId/:assignmentName')
+  @Roles(Role.STUDENT, Role.TA, Role.PROFESSOR)
+  async getStudentAssignmentProgress(
+    @Param('userId') userId: number,
+    @Param('courseId') courseId: number,
+    @Param('assignmentName') assignmentName: string,
+  ): Promise<StudentAssignmentProgress | null> {
```

Comment:
ah i guess i was wrong, i thought i did make these endpoints with @Res whoops. I guess it's been so long i forgot lol.

I might refactor it later when i add the other endpoints since right now it doesn't throw any errors outside of the ones in the guards, and I think it looks pretty clean the way it is right now.